leakcanary-android-core/src/main/java/leakcanary/internal/tv/TvOnRetainInstanceListener.kt

```diff
@@ -0,0 +1,68 @@
+package leakcanary.internal.tv
+
+import android.app.Application
+import android.os.Handler
+import android.os.Looper
+import android.widget.Toast
+import leakcanary.RetainInstanceChange
+import leakcanary.LeakCanary
+import leakcanary.OnRetainInstanceListener
+import leakcanary.RetainInstanceChange.CountChanged
+import leakcanary.RetainInstanceChange.Reset
+
+/**
+ * [OnRetainInstanceListener] implementation for Android TV devices which keeps track of the current
+ * retained instances count and displays a helpful Toast message on current state of it.
+ * It will notify user when heap dump is going to happen, so that users who set the
+ * [LeakCanary.Config.retainedVisibleThreshold] to any value other than 1 will be aware of retain
+ * instances changes. If user set threshold for heap dump to 1 then this listener becomes no-op,
+ * as any retain instance change will trigger the heap dump with its own toast.
```

Comment:
I think we'll want to say "hey I'm dumping the heap" (and have a default impl that's showing the toast)

Reply:
There will be a standard toast telling user that the heap is being dumped. That's why I made this one no-op for that case - so it doesn't show two toasts